### ISSUE TITLE
[Dialog] (minor) focus trap behavior is inconsistent

### ISSUE THREAD

**Issue by akdetrick** [narmi/design_system#842]
### Component/Module
The `Dialog` focus trap is potentially affected by the presence of `Tabs` with tabs panels on the page.

### Screenshots (if applicable)
![Screen Shot 2022-11-04 at 9 27 56 AM](https://user-images.githubusercontent.com/231252/199989761-a2d34e83-18a1-4b61-809c-0737ebf3e67c.png)

### Reproduction Steps
Put a `Dialog` on a page with `Tabs` containing tabs panels.

When tab panels are present in `banking`, the focus outline is not visible on the close button.
On other pages, the default focus moves to the close button and the focus outline is visible.

In isolation, this doesn't seem to be an issue: https://narmi.github.io/design_system/iframe.html?args=&id=components-dialog--focus-management&viewMode=story

### Additional Details / Context
We most likely want to try swapping out the library we're using for the focus trap. There are some other hooks that have much better features anyway, including a way to specify default focus in a declarative way on the consumer side via data attributes.

**Comment by edakrong** [comment]
@akdetrick @wrbrand @malindho Whats timing on prioritizing this issue?

**Comment by akdetrick** [comment]
Adding to a March milestone. It isn't an urgent fix, but we should get this out before making any breaking changes for the next major NDS version

**Comment by akdetrick** [comment]
Fixed by: https://github.com/narmi/design_system/pull/982

All products are upgraded to a version later than the `2.52.2` release which includes the fix.
